pkg/operator/controller/dns/controller.go

```diff
@@ -580,6 +646,37 @@ func (r *reconciler) ToDNSRecords(ctx context.Context, o client.Object) []reconc
 	return requests
 }
 
+func (r *reconciler) MapOnRecordDelete(ctx context.Context, o client.Object) []reconcile.Request {
+	deletedRecord, ok := o.(*iov1.DNSRecord)
+	if !ok {
+		log.Info("failed to read DNS record")
```

Comment:
```suggestion
		log.Infof("Got unexpected object; expected type DNSRecord, got type %T", o)
```

Reply:
Or something more like this:
```suggestion
		log.Error(nil, "Got unexpected type of object", "expected", "DNSRecord", "actual", fmt.Sprintf("%T", o))
```